### PR TITLE
arch: fix archinfo and misc for ARM

### PR DESCRIPTION
**Pull request by jaimesouza**
This PR fixes some issues in archinfo.py and misc.py that generated errors when running on ARM (Isambard).

### CODE REVIEW

File: devito/passes/iet/misc.py [hunk 1]
```diff
@@ -18,7 +19,7 @@ def avoid_denormals(iet):
     are normally flushed when using SSE-based instruction sets, except when
     compiling shared objects.
     """
-    if iet.is_ElementalFunction:
+    if iet.is_ElementalFunction or isinstance(archinfo.get_platform(), archinfo.Arm):
```
Comment (FabioLuporini) [inline devito/passes/iet/misc.py:22]: The compilation passes (`iet_pass`) are independent of the underlying architecture.

If you don't want `avoid_denormals` on Arm (even though this appears to be a mere work around) the way to go is to create a suitable compilation pipeline for Arm, which is identical to Intel except for the lack of `avoid_denormals`.

There already exists one such Arm compilation pipeline: https://github.com/devitocodes/devito/blob/master/devito/core/__init__.py#L31 -- as you can see, named `ArmOperator`.

You should go to ArmOperator, intercept (override) `_specialize_iet` so as to drop `avoid_denormals`

**Review by FabioLuporini — CHANGES_REQUESTED** [review]
@rhodrin maybe time to add an ARM action to our CI workflows?

**Review by FabioLuporini — APPROVED** [review]
LGTM. 

@georgebisbas can you try this patch?

@rhodrin can we add a free GH Actions runner?

**Comment by georgebisbas** [comment]
> @georgebisbas can you try this patch?

Yes

**Comment by georgebisbas** [comment]
I can confirm that the patch works and I could execute
acoustic and elastic operator.

**Comment by mloubout** [comment]
>can we add a free GH Actions runner?

You mean you want a `CI-arm`? That easily doable or can be added to `CI-cpu` but there is no ARM chips on GH actions.

**Comment by georgebisbas** [comment]
> > can we add a free GH Actions runner?
> 
> You mean you want a `CI-arm`? That easily doable or can be added to `CI-cpu` but there is no ARM chips on GH actions.

@mloubout : https://github.blog/changelog/2019-12-03-github-actions-self-hosted-runners-on-arm-architectures/
is that of use?

**Comment by georgebisbas** [comment]
Closed as superseded by #1515

**Pull request closed without merging**